File: CS_test_call/test_valuation.cs
using System;
using System.Globalization;

class Program
{
    static void Main(string[] args)
    {
        // Grab the ID sent from PHP
        string cadastralId = args.Length > 0 ? args[0] : "UNKNOWN-LOT";

        // Generate the random price
        Random random = new Random();
        double randomPrice = random.NextDouble() * (2500000.0 - 150000.0) + 150000.0;
        randomPrice = Math.Round(randomPrice, 2);

        // Manually format the string as JSON (No external libraries needed!)
        string jsonOutput = string.Format(CultureInfo.InvariantCulture,
            "{{\"status\":\"success\",\"cadastral_id\":\"{0}\",\"estimated_value\":{1}}}", 
            cadastralId, randomPrice);

        // Print to PHP
        Console.WriteLine(jsonOutput);
    }
}

File: Py_test_call/test_valuation.py
import sys
import json
import random

# Grab the ID sent from PHP (or use a fallback if missing)
cadastral_id = sys.argv[1] if len(sys.argv) > 1 else "UNKNOWN-LOT"

# Generate a random home price (e.g., between $150k and $2.5m)
# round(..., 2) ensures it formats like normal currency
random_price = round(random.uniform(150000.00, 2500000.00), 2)

# Package the result into a dictionary
result = {
    "status": "success",
    "cadastral_id": cadastral_id,
    "estimated_value": random_price
}

# Print the JSON string (This is what PHP will capture)
print(json.dumps(result))